### ISSUE TITLE
Failing opam install

### ISSUE THREAD

**Issue by A-Manning** [ocaml/dune#1125]
Trying to install dune on a system with Ubuntu 18.04, opam 1.2.2, OCaml 4.06.1

`$ opam install dune`
```
The following actions will be performed:
  ∗  install dune     1.1.1     
  ∗  install jbuilder transition
===== ∗  2 =====
Do you want to continue ? [Y/n] Y

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[dune] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of dune failed at "./boot.exe --release -j 4".

#=== ERROR while installing dune.1.1.1 ========================================#
# opam-version 1.2.2
# os           linux
# command      ./boot.exe --release -j 4
# path         ~/.opam/4.06.1/build/dune.1.1.1
# compiler     4.06.1
# exit-code    1
# env-file     ~/.opam/4.06.1/build/dune.1.1.1/dune-53118-6d07ae.env
# stdout-file ~/.opam/4.06.1/build/dune.1.1.1/dune-53118-6d07ae.out
# stderr-file  ~/.opam/4.06.1/build/dune.1.1.1/dune-53118-6d07ae.err
### stderr ###
# [...]
# Error: Error during linking
# -> required by src/configurator/flags/mk.exe
# -> required by src/configurator/flags/flags.sexp
# -> required by src/configurator/.merlin
# -> required by src/configurator/.merlin-exists
# -> required by src/configurator/.configurator.objs/configurator.ml.d
# -> required by src/configurator/.configurator.objs/configurator.ml.all-deps
# -> required by src/configurator/configurator.a
# -> required by _build/install/default/lib/dune/configurator/configurator.a
# -> required by dune.install



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions were aborted
  ∗  install jbuilder transition
The following actions failed
  ∗  install dune 1.1.1
No changes have been performed
```

**Comment by A-Manning** [comment]
```
~ $ opam source dune
Downloading archive of dune.1.1.1...
[dune] Archive in cache
Successfully extracted to ~/dune.1.1.1
```
```
dune.1.1.1 $ make release
make: *** No rule to make target 'release'.  Stop.
```

**Comment by A-Manning** [comment]
Tried building from source:
`git clone https://github.com/ocaml/dune.git && cd dune`
`make release`

```
ocaml bootstrap.ml
'~/.opam/4.06.1/bin/ocamllex.opt' -q src/meta_lexer.mll
'~/.opam/4.06.1/bin/ocamllex.opt' -q src/dune_lexer.mll
'~/.opam/4.06.1/bin/ocamllex.opt' -q src/dsexp/dune_lexer.mll
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/action.ml src/action_exec.ml src/action_intf.ml src/arg_spec.ml src/artifacts.ml src/bin.ml src/binary_kind.ml src/blang.ml src/build.ml src/build_interpret.ml src/build_system.ml src/clflags.ml src/cm_kind.ml src/colors.ml src/compilation_context.ml src/config.ml src/context.ml src/dep_path.ml src/dir_contents.ml src/dune_env.ml src/dune_file.ml src/dune_fmt.ml src/dune_lexer.ml src/dune_project.ml src/dynlink_supported.ml src/env.ml src/errors.ml src/exe.ml src/file_tree.ml src/findlib.ml src/gen_meta.ml src/gen_rules.ml src/glob_lexer.boot.ml src/import.ml src/inline_tests.boot.ml src/install.ml src/install_rules.ml src/installed_dune_file.ml src/jbuild_load.ml src/js_of_ocaml_rules.boot.ml src/lib.ml src/lib_deps_info.ml src/lib_rules.ml src/log.ml src/main.ml src/menhir.boot.ml src/merlin.ml src/meta.ml src/meta_lexer.ml src/ml_kind.ml src/mode.ml src/module.ml src/module_compilation.ml src/ocaml_flags.ml src/ocaml_version.ml src/ocamldep.ml src/odoc.boot.ml vendor/boot/opamBaseParser.ml vendor/boot/opamLexer.ml vendor/boot/opamParserTypes.ml src/opam_file.ml src/ordered_set_lang.ml src/package.ml src/path_dsexp.ml src/per_item.ml src/pform.ml src/preprocessing.ml src/print_diff.ml src/process.ml src/promotion.ml vendor/boot/re.ml src/report_error.ml src/scheduler.ml src/scope.ml src/setup.boot.ml src/simple_rules.ml src/stanza.ml src/string_with_vars.ml src/sub_system.ml src/sub_system_intf.ml src/sub_system_name.ml src/super_context.ml src/syntax.ml src/top_closure.ml src/utils.ml src/utop.ml src/value.ml src/variant.ml src/versioned_file.ml src/vfile_kind.ml src/watermarks.ml src/workspace.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/stdune/caml/caml.ml src/stdune/caml/result.ml src/stdune/caml/result_compat.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/dsexp/dsexp.ml src/dsexp/atom.ml src/dsexp/dune_lexer.ml src/dsexp/escape.ml src/dsexp/jbuild_lexer.boot.ml src/dsexp/lexer.ml src/dsexp/lexer_shared.ml src/dsexp/template.ml src/dsexp/types.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/fiber/fiber.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/ocaml-config/ocaml_config.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/stdune/stdune.ml src/stdune/ansi_color.ml src/stdune/array.ml src/stdune/bool.ml src/stdune/bytes.ml src/stdune/char.ml src/stdune/common.ml src/stdune/comparable.ml src/stdune/either.ml src/stdune/escape.ml src/stdune/exn.ml src/stdune/filename.ml src/stdune/float.ml src/stdune/fmt.ml src/stdune/hashable.ml src/stdune/hashtbl.ml src/stdune/hashtbl_intf.ml src/stdune/import.ml src/stdune/int.ml src/stdune/interned.ml src/stdune/io.ml src/stdune/list.ml src/stdune/loc.ml src/stdune/map.ml src/stdune/map_intf.ml src/stdune/option.ml src/stdune/or_exn.ml src/stdune/ordering.ml src/stdune/path.ml src/stdune/pp.ml src/stdune/result.ml src/stdune/set.ml src/stdune/set_intf.ml src/stdune/sexp.ml src/stdune/sexp_intf.ml src/stdune/staged.ml src/stdune/string.ml src/stdune/univ_map.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/which_program/which_program.boot.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamldep.opt' -modules src/xdg/xdg.ml > boot-depends.txt
'~/.opam/4.06.1/bin/ocamllex.opt' -q src/let-syntax/lexer.mll
'~/.opam/4.06.1/bin/ocamlc.opt' -g -w -40 -o boot.exe unix.cma boot.ml
./boot.exe --release
    ocamlopt src/let-syntax/pp.exe (exit 2)
(cd _build/default && ~/.opam/4.06.1/bin/ocamlopt.opt -w -40 -g -o src/let-syntax/pp.exe src/let-syntax/.pp.eobjs/lexer.cmx src/let-syntax/.pp.eobjs/pp.cmx)
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(roots.o): relocation R_X86_64_32 against symbol `caml_frametable' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals.o): relocation R_X86_64_32S against undefined symbol `caml_pending_signals' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals_asm.o): relocation R_X86_64_32S against symbol `caml_system__code_end' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(major_gc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(minor_gc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(memory.o): relocation R_X86_64_32 against symbol `caml_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(alloc.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compare.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(ints.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(floats.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(str.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(io.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(extern.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(intern.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(sys.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(gc_ctrl.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(obj.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(lexing.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(unix.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(callback.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(weak.o): relocation R_X86_64_32 against symbol `caml_ephe_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compact.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(finalise.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(custom.o): relocation R_X86_64_32 against symbol `caml_custom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(globroots.o): relocation R_X86_64_32 against symbol `caml_global_roots' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(backtrace.o): relocation R_X86_64_32 against symbol `caml_backtrace_last_exn' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(bigarray.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup_aux.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup.o): relocation R_X86_64_32S against symbol `caml_data_segments' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(fail.o): relocation R_X86_64_32 against symbol `caml_exn_Failure' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(misc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(freelist.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(hash.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(parsing.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(printexc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking
-> required by src/let-syntax/pp.exe
-> required by bin/main.pp.ml
-> required by bin/.main.objs/main.pp.ml.d
-> required by bin/.main.objs/main.pp.ml.all-deps
-> required by bin/main.a
-> required by bin/main_dune.exe
-> required by _build/install/default/bin/dune
-> required by dune.install
    ocamlopt src/configurator/flags/mk.exe (exit 2)
(cd _build/default && ~/.opam/4.06.1/bin/ocamlopt.opt -w -40 -g -o src/configurator/flags/mk.exe src/configurator/flags/.mk.eobjs/mk.cmx)
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(roots.o): relocation R_X86_64_32 against symbol `caml_frametable' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals.o): relocation R_X86_64_32S against undefined symbol `caml_pending_signals' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals_asm.o): relocation R_X86_64_32S against symbol `caml_system__code_end' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(major_gc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(minor_gc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(memory.o): relocation R_X86_64_32 against symbol `caml_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(alloc.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compare.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(ints.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(floats.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(str.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(array.o): relocation R_X86_64_32 against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(io.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(extern.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(intern.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(sys.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(gc_ctrl.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(obj.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(unix.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(callback.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(weak.o): relocation R_X86_64_32 against symbol `caml_ephe_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compact.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(finalise.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(custom.o): relocation R_X86_64_32 against symbol `caml_custom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(globroots.o): relocation R_X86_64_32 against symbol `caml_global_roots' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(backtrace.o): relocation R_X86_64_32 against symbol `caml_backtrace_last_exn' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(bigarray.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup_aux.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup.o): relocation R_X86_64_32S against symbol `caml_data_segments' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(fail.o): relocation R_X86_64_32 against symbol `caml_exn_Failure' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(misc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(freelist.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(hash.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(parsing.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(printexc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking
-> required by src/configurator/flags/mk.exe
-> required by src/configurator/flags/flags.sexp
-> required by src/configurator/.merlin
-> required by src/configurator/.merlin-exists
-> required by src/configurator/.configurator.objs/configurator.ml.d
-> required by src/configurator/.configurator.objs/configurator.ml.all-deps
-> required by src/configurator/configurator.a
-> required by _build/install/default/lib/dune/configurator/configurator.a
-> required by dune.install
Makefile:10: recipe for target 'release' failed
make: *** [release] Error 1
```

**Comment by diml** [comment]
This doesn't seem related to Dune. Can you produce any binary with the ocamlopt binary installed on your system? i.e. does the following work:

```sh
$ echo 'print_endline "Hello, world!"' > hello.ml
$ ocamlopt -o hello.exe hello.ml
```

**Comment by A-Manning** [comment]
I guess this is an ocamlopt issue then?
I get the following output:
```
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(roots.o): relocation R_X86_64_32 against symbol `caml_frametable' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(major_gc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(minor_gc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(memory.o): relocation R_X86_64_32 against symbol `caml_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compare.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(ints.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(floats.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(str.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(io.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(extern.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(intern.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(sys.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(gc_ctrl.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(obj.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(unix.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(callback.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(weak.o): relocation R_X86_64_32 against symbol `caml_ephe_ref_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(compact.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(finalise.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(custom.o): relocation R_X86_64_32 against symbol `caml_custom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(globroots.o): relocation R_X86_64_32 against symbol `caml_global_roots' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(backtrace.o): relocation R_X86_64_32 against symbol `caml_backtrace_last_exn' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(bigarray.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup_aux.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(startup.o): relocation R_X86_64_32S against symbol `caml_data_segments' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(fail.o): relocation R_X86_64_32 against symbol `caml_exn_Failure' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals.o): relocation R_X86_64_32S against undefined symbol `caml_pending_signals' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(signals_asm.o): relocation R_X86_64_32S against symbol `caml_system__code_end' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(misc.o): relocation R_X86_64_32 against `.rodata.str1.8' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(freelist.o): relocation R_X86_64_32S against `.bss' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(alloc.o): relocation R_X86_64_32S against undefined symbol `caml_atom_table' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(hash.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(parsing.o): relocation R_X86_64_32S against `.rodata' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: ~/.opam/4.06.1/lib/ocaml/libasmrun.a(printexc.o): relocation R_X86_64_32 against `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIC
/usr/bin/x86_64-linux-gnu-ld: final link failed: Nonrepresentable section on output
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking
```

**Comment by A-Manning** [comment]
Seems to have been fixed by `opam switch reinstall 4.06.1`, and it looks like it wasn't an issue with dune in the first place. Thanks for the help, closing.
